paddle/operators/add_op.h

```diff
@@ -29,8 +30,9 @@ class AddKernel : public framework::OpKernel {
 
     output->mutable_data<T>(context.GetPlace());
 
-    output->flat<T>().device(*(context.GetEigenDevice<Place>())) =
-        input0.flat<T>() + input1.flat<T>();
+    framework::EigenVector<T>::From(*output).device(
+        *(context.GetEigenDevice<Place>())) =
+        framework::EigenVector<T>(*input0) + framework::EigenVector<T>(*input1);
```

Comment:
Sure. I understand that we need Flatten. I just didn't have enough time today to add it. If would be great if you can take this PR and go on finishing it. Thanks!